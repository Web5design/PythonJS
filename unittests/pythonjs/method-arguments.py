class A:

    def method(self, a, b, c=3, d=4, *args, **kwargs):
        print a, b, c, d, len(args), len(kwargs)
        return a+b+c+d

print '>>> A().method(1,2)'
print A().method(1,2)
print '>>> A().method(11,22,33,44,55,66)'
print A().method(11,22,33,44,55,66)
print '>>> A().method(1, 2, e=True)'
print A().method(1, 2, e=True)
print '>>> A().method(11,22,33,44,55,66,e=True,f=True)'
print A().method(11,22,33,44,55,66,e=True,f=True)

