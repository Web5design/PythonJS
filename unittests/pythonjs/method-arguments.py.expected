>>> A().method(1,2)
1 2 3 4 0 0
10
>>> A().method(11,22,33,44,55,66)
11 22 33 44 2 0
110
>>> A().method(1, 2, e=True)
1 2 3 4 0 1
10
>>> A().method(11,22,33,44,55,66,e=True,f=True)
11 22 33 44 2 2
110
